src/transaction.js

```diff
@@ -435,10 +441,15 @@ Transaction.prototype.toBuffer = function (buffer, initialOffset) {
     writeVarSlice(txOut.script)
   })
 
-  if (hasWitnesses) {
+  if (serializeWitnesses) {
+    var isNull = true
     this.ins.forEach(function (input) {
       writeVector(input.witness)
+      isNull = isNull && input.witness.length === 0
     })
+    if (isNull) {
+      throw new Error('Transaction has superfluous witness data')
+    }
```

Comment:
`this.ins.every(function (input) { return input.witness.length === 0 })` is clearer IMHO